test/protocol-tests-core/src/main/java/software/amazon/awssdk/protocol/asserts/unmarshalling/UnmarshalledResultAssertion.java

```diff
@@ -77,4 +90,36 @@ private String getOutputClassName(UnmarshallingTestContext context) {
         return context.getModel().getOperations().get(context.getOperationName()).getReturnType()
                       .getReturnType();
     }
+
+    private static class CustomComparatorFactory extends ReflectionComparatorFactory {
+        private static ReflectionComparator getComparator() {
+            List<Comparator> comparators = new ArrayList<>();
+            comparators.add(new InstantComparator());
+            comparators.addAll(ReflectionComparatorFactory.getComparatorChain(Collections.emptySet()));
+            return new ReflectionComparator(comparators);
+        }
+    }
+
+    private static class InstantComparator implements Comparator {
+        @Override
+        public boolean canCompare(Object left, Object right) {
+            return left instanceof Instant || right instanceof Instant;
```

Comment:
Query :why isn't  it ?
```java
return left instanceof Instant && right instanceof Instant;
```

Reply:
I had the same thought at first. I looked at other comparator implementations and they did ||. It actually makes sense after I thought about it. This comparator knows whether two instants are equal, but it also knows whether one instant is equal to a non-instant, so it makes sense to say it can compare an instant and a non-instant (to return false).

Reply:
got it, thanks  👍